CMakeLists.txt

```diff
@@ -1,303 +1,159 @@
-        SET(CMAKE_EXE_LINKER_FLAGS      "-fprofile-arcs -ftest-coverage")
-        SET(CMAKE_SHARED_LINKER_FLAGS   "-fprofile-arcs -ftest-coverage")
-        SET(CMAKE_STATIC_LINKER_FLAGS   "-fprofile-arcs -ftest-coverage")
-    ENDIF(${WITH_COVERAGE})
-ENDIF(${UNIX})
-
-# OS Definitions
-IF(UNIX)
-    IF(APPLE)   #OSX
-        ADD_DEFINITIONS(-DOS_MAC)
-
-        SET(CMAKE_MACOSX_RPATH ON)
-        SET(CMAKE_SKIP_BUILD_RPATH  FALSE)
-        SET(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
-        SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/${AF_INSTALL_LIB_DIR}")
-        SET(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
-
-        LIST(FIND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES "${CMAKE_INSTALL_PREFIX}/${AF_INSTALL_LIB_DIR}" isSystemDir)
-        IF("${isSystemDir}" STREQUAL "-1")
-            SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/${AF_INSTALL_LIB_DIR}")
-        ENDIF("${isSystemDir}" STREQUAL "-1")
-    ELSE(APPLE) #Linux
-        ADD_DEFINITIONS(-DOS_LNX)
-    ENDIF()
-ELSE(${UNIX}) #Windows
-    ADD_DEFINITIONS(-DOS_WIN -DNOMINMAX)
-    IF(MSVC)
-        # MP is multiprocess compilation. Gm- disables minimal rebuilds
-        # http://stackoverflow.com/questions/6172205/how-can-i-do-a-parallel-build-in-visual-studio-2010vvvvvvvv
-        # http://www.kitware.com/blog/home/post/434
-        SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /MP /Gm- /bigobj")
-        SET(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS} /MP /Gm-")
-    ENDIF(MSVC)
-ENDIF()
-
-# Architechture Definitions
-INCLUDE(TargetArch)
-target_architecture(ARCH)
-
-INCLUDE(Version)
-
-IF(${BUILD_CPU})
-    ADD_SUBDIRECTORY(src/backend/cpu)
-ENDIF()
-
-IF(${BUILD_CUDA})
-    ADD_SUBDIRECTORY(src/backend/cuda)
-ENDIF()
-
-IF(${BUILD_OPENCL})
-    ADD_SUBDIRECTORY(src/backend/opencl)
-ENDIF()
-
-IF(${BUILD_UNIFIED})
-    ADD_DEFINITIONS(-DAF_UNIFIED)
-    ADD_SUBDIRECTORY(src/api/unified)
-ENDIF()
-
-IF(${BUILD_DOCS})
-    ADD_SUBDIRECTORY(docs)
-ENDIF()
-
-ADD_EXECUTABLE(bin2cpp ${PROJECT_SOURCE_DIR}/CMakeModules/bin2cpp.cpp)
-
-IF(${BUILD_TEST})
-    ENABLE_TESTING()
-    ADD_SUBDIRECTORY(test)
-ENDIF()
-
-IF(${BUILD_EXAMPLES})
-    ADD_SUBDIRECTORY(examples)
-ENDIF()
-
-##
-# Installation of headers, and CMake scripts
-##
-INSTALL(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/include/" DESTINATION "${AF_INSTALL_INC_DIR}"
-    COMPONENT headers
-    FILES_MATCHING
-    PATTERN "*.h"
-    PATTERN "*.hpp"
-    PATTERN ".gitignore" EXCLUDE
+configure_file(
+    ${PROJECT_SOURCE_DIR}/CMakeModules/version.hpp.in
+    ${PROJECT_BINARY_DIR}/version.hpp
 )
 
-## The ArrayFire version file is generated and won't be included above, install
-## it separately.
-INSTALL(FILES
-    ${PROJECT_SOURCE_DIR}/include/af/version.h DESTINATION "${AF_INSTALL_INC_DIR}/af/"
-    COMPONENT headers
-)
+add_executable(bin2cpp ${PROJECT_SOURCE_DIR}/CMakeModules/bin2cpp.cpp)
+target_compile_features(bin2cpp
+  PRIVATE
+    cxx_raw_string_literals
+  )
```

Comment:
Some compilers do not support all features in a standard. The -std=c++11 flag will enable the best support possible for a particular standard. This, on the other hand, will test a compiler if a certain feature is supported and if not it will throw an error message.